src/lis.py

```diff
@@ -47,16 +47,15 @@ def longest_increasing_substring(x: Sequence[Any]) -> tuple[int, int]:
     >>> longest_increasing_substring([12, 45, 32, 65, 78, 23, 35, 45, 57])
     (5, 9)
     """
-    for i in range(len(x)+1):
+    y=0
+
+    for i in range(1,len(x)):
         list1=[]
-        y=0
-        if is_increasing([x[i],x[i+1]])==True:
-            list1.append(i)
+        if is_increasing([x[i-1],x[i]])==True:
```

Comment:
Isn't this just a complicated way to test `x[i-1] <= x[i]`?

The `is_increasing` function is useful when you have a long sequence and you want to check it, but if you just want to know if one number is less than or equal to another, then you might as well use the `<=` operator.

Also, you never need to check `if expr == True` for an `if`-statement. The `== True` isn't necessary; you can just write `if expr:`.